crates/language/src/language.rs

```diff
@@ -201,6 +201,7 @@ pub struct CachedLspAdapter {
     pub name: LanguageServerName,
     pub disk_based_diagnostic_sources: Vec<String>,
     pub disk_based_diagnostics_progress_token: Option<String>,
+    pub previous_document_diagnostic_result_id: Option<String>,
```

Comment:
Still not sure about the place to store this token that Zed should store and use between sending/receiving `textDocument/diagnostic` commands.

Reply:
Moved the `previous_document_diagnostic_result_id` to `LanguageServerState` enum. Let me know if that's ok.